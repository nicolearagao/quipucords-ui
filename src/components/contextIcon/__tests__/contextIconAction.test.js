import React from 'react';
import { ContextIconAction, ContextIconActionVariant } from '../contextIconAction';

describe('ContextIconAction Component', () => {
  it('should render a basic component with default props', async () => {
    const component = await shallowComponent(<ContextIconAction />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should export, and handle icon variants', () => {
    expect(ContextIconActionVariant).toMatchSnapshot('variants');

    Object.entries(ContextIconActionVariant).forEach(([key, value]) => {
      const component = renderComponent(<ContextIconAction symbol={value} />);
      expect(component.props).toMatchSnapshot(`icon variant, ${key}`);
    });
  });
});
