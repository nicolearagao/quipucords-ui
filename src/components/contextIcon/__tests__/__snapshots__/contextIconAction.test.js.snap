// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, canceled 1`] = `
{
  "children": [],
  "symbol": "failed",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, cancelled 1`] = `
{
  "children": [],
  "symbol": "failed",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, completed 1`] = `
{
  "children": [],
  "symbol": "completed",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, created 1`] = `
{
  "children": [],
  "symbol": "running",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, failed 1`] = `
{
  "children": [],
  "symbol": "failed",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, paused 1`] = `
{
  "children": [],
  "symbol": "paused",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, pending 1`] = `
{
  "children": [],
  "symbol": "pending",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, play 1`] = `
{
  "children": [],
  "symbol": "play",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: icon variant, running 1`] = `
{
  "children": [],
  "symbol": "running",
}
`;

exports[`ContextIconAction Component should export, and handle icon variants: variants 1`] = `
{
  "canceled": "failed",
  "cancelled": "failed",
  "completed": "completed",
  "created": "running",
  "failed": "failed",
  "paused": "paused",
  "pending": "pending",
  "play": "play",
  "running": "running",
}
`;

exports[`ContextIconAction Component should render a basic component with default props: basic 1`] = `
<PlayIcon
  color="currentColor"
  noVerticalAlign={false}
  size="sm"
/>
`;
